omnisafe/envs/meta_drive_env.py

```diff
@@ -0,0 +1,194 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Environments interface of SafeMetaDrive."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, ClassVar
+
+import numpy as np
+import torch
+
+from omnisafe.common.logger import Logger
+from omnisafe.envs.core import CMDP, env_register
+from omnisafe.typing import DEVICE_CPU
+
+
+META_DRIVE_AVAILABLE = True
+try:
+    from metadrive import SafeMetaDriveEnv
+except ImportError:
+    META_DRIVE_AVAILABLE = False
+
+
+@env_register
+class SafetyMetaDriveEnv(CMDP):
+    """SafeMetaDrive Environment.
+
+    More information about MetaDrive environment is provided in https://github.com/metadriverse/metadrive.
+    For the details of environment configuration, please refer to https://github.com/decisionforce/EGPO.
+
+    Args:
+        env_id (str): Environment id.
+        num_envs (int, optional): Number of environments. Defaults to 1.
+        device (torch.device, optional): Device to store the data. Defaults to
+            ``torch.device('cpu')``.
+
+    Keyword Args:
+        meta_drive_config (dict, optional): MetaDrive configuration, containing following keys:
+            - ``horizon``: Max iterations of interactions.
+            - ``random_traffic``: Whether to use random traffic.
+            - ``crash_vehicle_penalty``: The penalty when crash into other vehicles.
+            - ``crash_object_penalty``: The penalty when crash into other objects.
+            - ``out_of_road_penalty``: The penalty when out of road.
+
+    Attributes:
+        need_auto_reset_wrapper (bool): Whether to use auto reset wrapper.
+        need_time_limit_wrapper (bool): Whether to use time limit wrapper.
+    """
+
+    need_auto_reset_wrapper: bool = True
+    need_time_limit_wrapper: bool = False
+    env_spec_log: dict[str, Any]
+
+    _support_envs: ClassVar[list[str]] = [
+        'SafeMetaDrive',
+    ]
+
+    def __init__(
+        self,
+        env_id: str,
+        num_envs: int = 1,
+        device: torch.device = DEVICE_CPU,
+        **kwargs: Any,  # pylint: disable=unused-argument
+    ) -> None:
+        """Initialize an instance of :class:`SafetyGymnasiumEnv`."""
+        super().__init__(env_id)
+        self._num_envs = num_envs
+        self._device = torch.device(device)
+
+        if META_DRIVE_AVAILABLE:
+            self._env = SafeMetaDriveEnv(config=kwargs.get('meta_drive_config'))
+        else:
+            raise ImportError('Please install MetaDrive to use Safe SafeMetaDrive!')
```

Comment:
```suggestion
            raise ImportError('Please install MetaDrive to use SafeMetaDrive!')
```

Reply:
Can we further recommend the command, such as `pip install metadrive`?

Reply:
It is a pretty nice idea! I have made a revision.